Source/Particles/Collision/BinaryCollision/ProtonBoronFusionInitializeMomentum.H

```diff
@@ -22,35 +27,227 @@ namespace {
+        // single precision.
+#ifdef AMREX_SINGLE_PRECISION_PARTICLES
+        const amrex::ParticleReal ma_sq = m_alpha*m_alpha;
+        const amrex::ParticleReal mBe_sq = m_beryllium*m_beryllium;
+#else
+        constexpr amrex::ParticleReal ma_sq = m_alpha*m_alpha;
+        constexpr amrex::ParticleReal mBe_sq = m_beryllium*m_beryllium;
+#endif
+
+        // Normalized momentum of colliding particles (proton and boron)
+        const amrex::ParticleReal u1x = soa_1.m_rdata[PIdx::ux][idx_1];
+        const amrex::ParticleReal u1y = soa_1.m_rdata[PIdx::uy][idx_1];
+        const amrex::ParticleReal u1z = soa_1.m_rdata[PIdx::uz][idx_1];
+        const amrex::ParticleReal u2x = soa_2.m_rdata[PIdx::ux][idx_2];
+        const amrex::ParticleReal u2y = soa_2.m_rdata[PIdx::uy][idx_2];
+        const amrex::ParticleReal u2z = soa_2.m_rdata[PIdx::uz][idx_2];
+
+        // Compute Lorentz factor gamma in the lab frame
+        const amrex::ParticleReal g1 = std::sqrt( 1._prt + (u1x*u1x+u1y*u1y+u1z*u1z)*inv_csq );
+        const amrex::ParticleReal g2 = std::sqrt( 1._prt + (u2x*u2x+u2y*u2y+u2z*u2z)*inv_csq );
+
+        // Compute momenta
+        const amrex::ParticleReal p1x = u1x * m1;
+        const amrex::ParticleReal p1y = u1y * m1;
+        const amrex::ParticleReal p1z = u1z * m1;
+        const amrex::ParticleReal p2x = u2x * m2;
+        const amrex::ParticleReal p2y = u2y * m2;
+        const amrex::ParticleReal p2z = u2z * m2;
+        // Square norm of the total (sum between the two particles) momenta in the lab frame
+        const amrex::ParticleReal p_total_sq =  std::pow(p1x+p2x, 2) +
+                                                std::pow(p1y+p2y, 2) +
+                                                std::pow(p1z+p2z, 2);
+
+        // Total energy of proton+boron in the lab frame
+        const amrex::ParticleReal E_lab = (m1 * g1 + m2 * g2) * c_sq;
+        // Total energy squared of proton+boron in the center of mass frame, calculated using the
+        // Lorentz invariance of the four-momentum norm
+        const amrex::ParticleReal E_star_sq = E_lab*E_lab - c_sq*p_total_sq;
+        // Total energy squared of beryllium+alpha in the center of mass frame
+        // In principle, the term - E_rest_pb + E_rest_abe + E_fusion is not needed and equal to
+        // zero (i.e. the energy liberated during fusion is equal to the mass difference). However,
+        // due to possible inconsistencies in how the mass is defined in the code (e.g. currently,
+        // the mass of hydrogen is the mass of the proton, not including the electron, while the
+        // mass of the other elements is the atomic mass, that includes the electrons mass), it is
+        // probably more robust to subtract the rest masses and to add the fusion energy to the
+        // total kinetic energy.
+        const amrex::ParticleReal E_star_f_sq = std::pow(std::sqrt(E_star_sq)
```

Comment:
Not urgent, because it's not in a hot loop, so just as a comment:

Calculating the power2 based on a multiplication is faster than `pow(..., 2.)`.
You could write this in two `const` variables if you like to keep the square readable or we could define general math functions (square, quad, etc.). Not urgent, just a thought if we also need ^2 and ^3 in kernels.